pkg/controller/dnszone/dnszone_controller.go

```diff
@@ -236,21 +298,191 @@ func shouldSync(desiredState *hivev1.DNSZone) (bool, time.Duration) {
 	return false, delta
 }
 
-// getAWSClient generates an awsclient
-func (r *ReconcileDNSZone) getAWSClient(dnsZone *hivev1.DNSZone, dnsLog log.FieldLogger) (awsclient.Client, error) {
-	// This allows for using host profiles for AWS auth.
-	var secretName, regionName string
+func (r *ReconcileDNSZone) getActuator(dnsZone *hivev1.DNSZone, dnsLog log.FieldLogger) (Actuator, error) {
+	if dnsZone.Spec.AWS != nil && dnsZone.Spec.AWS.AccountSecret.Name != "" {
```

Comment:
It should be a separate error for the case where the spec stipulates AWS but the secret name is missing, rather than lumping the two together in a "unable to determine which actuator to use" error.

You probably don't even need to check that the name is not empty. If the name is empty, then the Get of the secret will throw an error anyway.